### PR TITLE
New Resource : google_dialogflow_conversation_profile

### PR DESCRIPTION
**Pull request by jcanizalez**
Hi team,

fixes hashicorp/terraform-provider-google#12466

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-the-terraform-providers), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-resource
`google_dialogflow_conversation_profile`
```

### CODE REVIEW

**Comment by jcanizalez** [comment]
hi @shuyama1 can you please share me the logs for the errors in tests? 
Thanks

**Comment by shuyama1** [comment]
Hi @jcanizalez, I don't think the failing test is related to this PR and `TestAccDialogflowConversationProfile_update` seems passed.

**Comment by jcanizalez** [comment]
awesome thank you @shuyama1

**Comment by jcanizalez** [comment]
Hi @shuyama1  I hope you are doing well, did you get chance to take a look 
Thanks

**Comment by shuyama1** [comment]
Hi @jcanizalez! Sorry for the delay on this. It seems that the tests for this resource are skipped. Are you able to test it locally?

File: mmv1/products/dialogflow/api.yaml [hunk 1]
```diff
@@ -334,58 +332,543 @@ objects:
...
+                            - !ruby/object:Api::Type::Boolean
+                              name: "dropVirtualAgentMessages"
+                              description: |
+                                If set to true, all messages from virtual agent are dropped.
+                            - !ruby/object:Api::Type::Boolean
+                              name: "dropIvrMessages"
+                              description: |
+                                If set to true, all messages from ivr stage are dropped.
+                        - !ruby/object:Api::Type::NestedObject
+                          name: "knowledgeBaseQuerySource"
```
Comment (shuyama1) [inline mmv1/products/dialogflow/api.yaml:544]: Based on the [API documentation](https://cloud.google.com/dialogflow/es/docs/reference/rest/v2/projects.conversationProfiles#ConversationProfile.SuggestionQueryConfig), it looks like Union field `query_source` can be only one of the following:
- knowledgeBaseQuerySource
- documentQuerySource
- dialogflowQuerySource
We need to add `conflicts` for these fields.

**Review by shuyama1 — COMMENTED** [review]
Thanks for adding the resource and sorry for the delay on reviewing!

Some general suggestions:
- For any nest objects, it would be nice to have at least one of the subfields required or have `at_least_one_of` set for all the sub fields to prevent users from configuring an empty block, which may cause more problems.
- For enum fields, it’d nice if we could implement them as strings and list all the possible values in the description, as setting a field as enum means code change is always inevitable if a new possible value is added to the field.

File: mmv1/products/dialogflow/api.yaml [hunk 2]
```diff
@@ -334,58 +332,543 @@ objects:
...
+      - !ruby/object:Api::Type::NestedObject
+        name: "humanAgentHandoffConfig"
+        description: |
+          Configuration for connecting to a live agent.
+          Currently, this feature is not general available, please contact Google to get access.
+        properties:
+          - !ruby/object:Api::Type::NestedObject
+            name: "livePersonConfig"
```
Comment (shuyama1) [inline mmv1/products/dialogflow/api.yaml:750]: It seems that only of the `livePersonConfig` and `salesforceLiveAgentConfig` can be added. We may also need to add `conflicts` to these two fields.

**Pull request closed without merging**